src/IntegrationUtils.ts

```diff
@@ -142,8 +142,13 @@ const choiceList = [
 					// do nothing with config-map or secret
 					break;
 				case vscodeTasksIntegration:
-					await handleDefinedTask(context);
-					resolve();
+					await handleDefinedTask(context).then(() => {
```

Comment:
That part doesn't look right. You are mixing async/await with promises. Its safer to only use one of it. I'd suggest to switch to something like: 
`
try {
				await handleDefinedTask(context);
				resolve();
			     } catch( (err) => {
				reject(err);
				errorEncountered = true;
			     };
`

Reply:
I wanted to stay coherent with the other parts of the function.

What do you think to create a specific refactor task to handle that for the whole file and to handle it in another Pull Request?

Reply:
You introduced it in this PR, so it would be good to fix it in here.

Reply:
I'm not introducing the new notation in this PR.
This way of writing it is used several times in the same method. The idea is to stay coherent with other parts of the code. Even if not optimal it stays more readable than mixing styles inside the same method.

Reply:
it is used in the same method at these places:
- https://github.com/camel-tooling/vscode-camelk/blob/master/src/IntegrationUtils.ts#L82
- https://github.com/camel-tooling/vscode-camelk/blob/99fc0f5c0c16c82d17d5b57126dd34092bf6f9dd/src/IntegrationUtils.ts#L94
- https://github.com/camel-tooling/vscode-camelk/blob/99fc0f5c0c16c82d17d5b57126dd34092bf6f9dd/src/IntegrationUtils.ts#L106
- https://github.com/camel-tooling/vscode-camelk/blob/99fc0f5c0c16c82d17d5b57126dd34092bf6f9dd/src/IntegrationUtils.ts#L118
- https://github.com/camel-tooling/vscode-camelk/blob/99fc0f5c0c16c82d17d5b57126dd34092bf6f9dd/src/IntegrationUtils.ts#L130

Reply:
But it wasn't in the position you changed. There you introduced it.

![image](https://user-images.githubusercontent.com/435853/80969705-54218c00-8e1a-11ea-9b30-354618fdece3.png)

Reply:
I ensured the code to stay coherent

Reply:
I created https://issues.redhat.com/browse/FUSETOOLS2-421 and try to get started to refactor to be coherent at the file level, there are some parts I'm really not confident to touch, for instance:

https://github.com/camel-tooling/vscode-camelk/blob/99fc0f5c0c16c82d17d5b57126dd34092bf6f9dd/src/IntegrationUtils.ts#L244-L290

https://github.com/camel-tooling/vscode-camelk/blob/99fc0f5c0c16c82d17d5b57126dd34092bf6f9dd/src/IntegrationUtils.ts#L292-L322

Reply:
A good rule of thumb is to always try to leave the code cleaner than you found it.

Reply:
I am fine if you want to tackle that in a different PR.

Reply:
the effort to do it is important. The risk of regression is high. It will more than double the size of the PR. This is why I really think it should be handled in another PR. That's great that it has been identified.